### PR TITLE
2599 oxford2 supports attestation

### LINKED ISSUE #2599: Change `endorsement` into `attestation`
Protocol Oxford introduces a renaming of all things related to `endorsement` or `endorsing` into `attestation` or `attesting`.

Do a run through of Taquito and update the names accordingly in:
- the codebase (types, rpc responses, constants, etc)
- the documentation (website, docs)
- the Taquito test dapp (if any)

### PR DESCRIPTION
**Pull request by hui-an-yang**
supported new name attestation along with endorsement(default but deprecated in oxford2)
[breaking change] local-forging now can only forge and parse attestation operation (no endorsement)

closes #2599 

Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have run the linter against the changes
- [ ] You have added unit tests (if relevant/appropriate)
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

In this PR, please also make sure: 

- [ ] You have linked this PR to the issue by putting `closes #TICKETNUMBER` in the description box (when applicable)
- [ ] You have added a concise description on your changes
## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: integration-tests/rpc-nodes.spec.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ CONFIGS().forEach(
   }) => {
     const Tezos = lib;
     const unrestrictedRPCNode = rpc.endsWith("ecadinfra.com") ? test.skip : test;
-    const oxfordAndAlpha = ProtoGreaterOrEqual(protocol, Protocols.ProxfordY) ? test : test.skip
+    const unrestrictedOxfordAndAlpha = ProtoGreaterOrEqual(protocol, Protocols.ProxfordY) ? test : test.skip;
```
[the remark this answers is not in the document]
Comment (hui-an-yang) [reply integration-tests/rpc-nodes.spec.ts:24]: line24 should rename to `oxfordAndAlpha`, will update in next commit
answering unrestricted is for bakingRights, endorsingRights, attestationRights that usually are not accessible through public rpc node, but we have made it able to test with rpc url `rpc.endsWith("ecadinfra.com")`

File: packages/taquito-local-forging/src/constants.ts [hunk 2]
```diff
@@ -236,7 +237,7 @@ export const kindMapping: { [key: number]: string } = {
   0x6c: 'transaction',
   0x6d: 'origination',
   0x06: 'ballot',
-  0x15: 'endorsement',
+  0x15: 'attestation',
```
[the remark this answers is not in the document]
Comment (hui-an-yang) [reply packages/taquito-local-forging/src/constants.ts:240]: In oxford2, baker can both `endorse` and `attest`, the limit in taquito is `kindMapping ` can only have one 0x15 in one protocol. The other option I thought about is letting user to specify protocol nairobi to be bale to forge endorsement and oxford to forge attestation only. Didn't end up doing it thinking if user going to update their code it's better migrate to something new than something deprecated and have to update code again in protocol P. Hope this brings clarity :)

[the remark this answers is not in the document]
Comment (dsawali) [reply packages/taquito-local-forging/src/constants.ts:240]: yeah I think we had this conversation before. Thanks for the refresher

**Review by dsawali — APPROVED** [review]
lgtm!

**Pull request merged**